### ISSUE TITLE
InfluxDB.Client.Linq -> Bad generated Query

### ISSUE THREAD

**Issue by deinok** [influxdata/influxdb-client-csharp#201]
Hi, 

I'm trying to generate a query using the LinQ system. And it is always empty.
Looks like it is generating something like this:

```
start_shifted = int(v: time(v: p4))
stop_shifted = int(v: time(v: p5)) + 1

from(bucket: p1) 
    |> range(start: time(v: start_shifted), stop: time(v: stop_shifted)) 
    |> filter(fn: (r) => (r["Id"] == p3)) 
    |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value") 
    |> drop(columns: ["_start", "_stop", "_measurement"]) 
    |> group()
```

the issue looks like it is coming from the `range()` function.

Insted of:
`|> range(start: time(v: start_shifted), stop: time(v: stop_shifted))`
I think it should generate:
`|> range(start: start_shifted, stop: stop_shifted) `

**Comment by bednar** [comment]
Hi @deinok,

thanks for using our client. 

The conversion to time - `time(v: start_shifted)`- is required. The `start_shifted` and `stop_shifted` has `nanosecond` precision, but the [range() function](https://docs.influxdata.com/influxdb/cloud/reference/flux/stdlib/built-in/transformations/range/) takes `seconds` precision for `Integer` type.

You can debug the produced query via: 

```csharp
var query = (from s in InfluxDBQueryable<Sensor>.Queryable("my-bucket", "my-org", _queryApi)
        where s.SensorId == "id-1"
        where s.Value > 12
        where s.Timestamp > new DateTime(2019, 11, 16, 8, 20, 15, DateTimeKind.Utc)
        where s.Timestamp < new DateTime(2021, 01, 10, 5, 10, 0, DateTimeKind.Utc)
        orderby s.Timestamp
        select s)
    .Take(2)
    .Skip(2);
    
Console.WriteLine("==== Debug LINQ Queryable Flux output ====");
var influxQuery = ((InfluxDBQueryable<Sensor>) query).ToDebugQuery();
foreach (var statement in influxQuery.Extern.Body)
{
    var os = statement as OptionStatement;
    var va = os?.Assignment as VariableAssignment;
    var name = va?.Id.Name;
    var value = va?.Init.GetType().GetProperty("Value")?.GetValue(va.Init, null);

    Console.WriteLine($"{name}={value}");
}
Console.WriteLine();
Console.WriteLine(influxQuery._Query);
```

- https://github.com/influxdata/influxdb-client-csharp/tree/master/Client.Linq

Regards

**Comment by deinok** [comment]
```
start_shifted = int(v: time(v: p2))

from(bucket: p1) 
    |> range(start: time(v: start_shifted)) 
    |> filter(fn: (r) => (r["Id"] == p3)) 
    |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value") 
    |> drop(columns: ["_start", "_stop", "_measurement"]) 
    |> group()
```

where:
p1: "default" - StringLiteral
p2: 0 - IntegralLiteral
p3: 346964633 - IntegralLiteral

So te issue looks like a bit like p3 is replaced as an integer insted of a String

My POCO structure is like:
```
	[Measurement(nameof(Engine))]
	public class Engine
	{

		[Column(nameof(ActiveEnergy))]
		public decimal ActiveEnergy { get; set; }

		[Column(nameof(EnergyEfficiency))]
		public decimal EnergyEfficiency { get; set; }

		[Column(nameof(ReactiveEnergy))]
		public decimal ReactiveEnergy { get; set; }

		[Column(nameof(TotalEnergy))]
		public decimal TotalEnergy { get; set; }

		[Column(IsTag = true)]
		public int Id { get; set; }

		[Column(IsTag = true)]
		public string Tag { get; set; }

		[Column(IsTimestamp = true)]
		public DateTime Timestamp { get; set; }

	}
```

**Comment by bednar** [comment]
@deinok thanks for detailed investigation 👍

We will have to fix our `Query Visitor` to supports this type of coercion.

- https://github.com/influxdata/influxdb-client-csharp/blob/c9d1c3ba7520219d69925b7b70783bfe0b865cf1/Client.Linq/Internal/QueryVisitor.cs#L68

**Comment by deinok** [comment]
Thank @bednar , for now I will change the `int Id` to `string Id`

**Comment by bednar** [comment]
The issue is fixed in [1.19.0](https://github.com/influxdata/influxdb-client-csharp/milestone/22) milestone. You can use latest dev version by:

```console
dotnet add package InfluxDB.Client --version 1.19.0-dev.3159
```

btw, the `1.19.0` change how looks like query generated from LINQ - for more info see: https://github.com/influxdata/influxdb-client-csharp/tree/master/Client.Linq#time-series
